lib/elixir/lib/map.ex

```diff
@@ -468,9 +468,15 @@ defmodule Map do
   """
   @spec merge(map, map, (key, value, value -> value)) :: map
   def merge(map1, map2, callback) when is_function(callback, 3) do
-    :maps.fold fn k, v2, acc ->
-      update(acc, k, v2, fn(v1) -> callback.(k, v1, v2) end)
-    end, map1, map2
+    if map_size(map1) > map_size(map2) do
```

Comment:
I think it would be better to swap arguments before and not duplicate `:maps.fold/3` call:
```elixir
{map1, map2} = if map_size(map1) > map_size(map2), do: {map2, map1}, else: {map1, map2}
```